### PR TITLE
feat: allow compactor to split files covering 2ns

### PR DESCRIPTION
**Pull request by Joe-Blount**
Working on a larger PR, I encountered a situation where too much data landed in a 2ns time range, so the partition was put on the skip table rather than compacting.  It was a situation that previously resolved, so I didn't want to let the scenario result in skipping.  

So this PR allows the compactor to split files covering a 2ns, into two files each covering 1ns (which is the smallest possible).  A few test cases verifying the minimum split scenarios are updated, because 2ns files now split.

Also in some of the more general insta tests, some scenarios that previously put a partition on the skip table now don't.

helps https://github.com/influxdata/idpe/issues/17974

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: compactor/src/components/split_or_compact/large_files_to_split.rs [hunk 1]
```diff
@@ -41,22 +41,34 @@ pub fn compute_split_times_for_large_files(
         let min_time = file.min_time.get();
         let max_time = file.max_time.get();
 
+        // TODO: it would be nice to check if these files overlap (e.g. if they're multiple levels), and
+        // coordinate the split time across all the files, rather than deciding the split time for each file
+        // as if its the only file under consideration.
```
Comment (Joe-Blount) [inline compactor/src/components/split_or_compact/large_files_to_split.rs:46]: this comment is unrelated to the change in this PR.  I just noticed the opportunity for further improvement, so I'm adding a TODO so I remember it.

File: compactor/tests/layouts/single_timestamp.rs [hunk 2]
```diff
@@ -290,22 +302,317 @@ async fn many_medium_files_time_range_1() {
...
+    - "L0.?[100,100] 20ns       |L0.?|                                                                                    "
+    - "L0.?[101,101] 20ns                                                                                                 |L0.?|"
+    - "Committing partition 1:"
+    - "  Soft Deleting 10 files: L0.11, L0.12, L0.13, L0.14, L0.15, L0.16, L0.17, L0.18, L0.19, L0.20"
+    - "  Creating 20 files"
+    - "**** Simulation run 12, type=compact(FoundSubsetLessThanMaxCompactSize). 2 Input Files, 165mb total:"
+    - "L0                                                                                                                 "
+    - "L0.24[100,100] 11ns 15mb |-----------------------------------------L0.24------------------------------------------|"
+    - "L1                                                                                                                 "
+    - "L1.22[100,100] 10ns 150mb|-----------------------------------------L1.22------------------------------------------|"
```
Comment (NGA-TRAN) [inline compactor/tests/layouts/single_timestamp.rs:392]: Can you remind me whether same min and max are expected?

**Review by NGA-TRAN — APPROVED** [review]
Except a question,  the code and tests look good. One more output files in the 2 ns case.
I wonder if our algorithm leads to 1ns case that needs to split a lot and then as a consequence in skip_compaction? Hope not

File: compactor/tests/layouts/single_timestamp.rs (hunk 2 above)
Comment (Joe-Blount) [reply compactor/tests/layouts/single_timestamp.rs:392]: > Can you remind me whether same min and max are expected?

as of this PR, yes its expected. that's what a file covering 1ns looks like.
Splitting a file covering 2ns will inevitably result in two files with the same min/max because they only cover 1ns
